modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/internal/propfind_query_legacy.rb

```diff
@@ -71,69 +45,80 @@ def initialize(storage)
+                  log_message "Parsing XML response body"
+                  doc = Nokogiri::XML(response.body.to_s)
+                  Rails.logger.info "Parsing response body"
+                  result = doc.xpath("/d:multistatus/d:response").each_with_object({}) do |resource_section, hash|
+                    source_path = UrlBuilder.path(@storage.uri.path, "/remote.php/dav/files", @username)
+                    resource = CGI.unescape(resource_section.xpath("d:href").text.strip).gsub!(source_path, "")
 
-                  result[resource] = {}
+                    hash[resource] = {}
 
-                  # In future it could be useful to respond not only with found, but not found props as well
-                  # resource_section.xpath("d:propstat[d:status[text() = 'HTTP/1.1 404 Not Found']]/d:prop/*")
-                  resource_section.xpath("d:propstat[d:status[text() = 'HTTP/1.1 200 OK']]/d:prop/*").each do |node|
-                    result[resource][node.name.to_s] = node.text.strip
+                    # In future it could be useful to respond not only with found, but not found props as well
+                    # resource_section.xpath("d:propstat[d:status[text() = 'HTTP/1.1 404 Not Found']]/d:prop/*")
+                    resource_section.xpath("d:propstat[d:status[text() = 'HTTP/1.1 200 OK']]/d:prop/*").each do |node|
+                      hash[resource][node.name.to_s] = node.text.strip
+                    end
                   end
-                end
 
-                ServiceResult.success(result:)
-              in { status: 405 }
-                Util.error(:not_allowed, "Outbound request method not allowed", error_data)
-              in { status: 401 }
-                Util.error(:unauthorized, "Outbound request not authorized", error_data)
-              in { status: 404 }
-                Util.error(:not_found, "Outbound request destination not found", error_data)
-              else
-                Util.error(:error, "Outbound request failed", error_data)
+                  log_message "Response parsed found: #{result.inspect}"
+                  ServiceResult.success(result:)
+                in { status: 405 }
+                  log_response(response)
+                  Util.error(:not_allowed, "Outbound request method not allowed", error_data)
+                in { status: 401 }
+                  log_response(response)
+                  Util.error(:unauthorized, "Outbound request not authorized", error_data)
+                in { status: 404 }
+                  log_response(response)
+                  Util.error(:not_found, "Outbound request destination not found", error_data)
+                else
+                  Util.error(:error, "Outbound request failed", error_data)
+                end
               end
             end
-
             # rubocop:enable Metrics/AbcSize
+
+            def log_response(response)
+              log_message "Storage responded with a #{response.status} code."
```

Comment:
Cool, that's a great question. HTTPX almost always return a response. There's a special object in case of a non status response (think of a timeout  for example) that behaves kinda like one.

The thing here is that we are using pattern matching, so if the response don't have a status, it will match the else clause (where we don't try to log the response status).